core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1012,13 +1013,18 @@ object TestUtils extends Logging {
     }.mkString("\n")
   }
 
+  type CreateLogDirectoryFn = (File,String) => Try[File]
+  type CreateLogDirectoryOverride = CreateLogDirectoryFn => CreateLogDirectoryFn
+
   /**
    * Create new LogManager instance with default configuration for testing
    */
   def createLogManager(logDirs: Seq[File] = Seq.empty[File],
                        defaultConfig: LogConfig = LogConfig(),
                        cleanerConfig: CleanerConfig = CleanerConfig(enableCleaner = false),
-                       time: MockTime = new MockTime()): LogManager = {
+                       time: MockTime = new MockTime(),
+                       createLogDirectoryOverride: CreateLogDirectoryOverride = original => original
```

Comment:
Since this is really a one-off need, it would be nice not to clutter up `TestUtils`. Would it be possible to construct the regular `LogManager` object in `testCreateLogWithLogDirFallback` and then use a Mockito spy to alter the behavior of `createLogDirectory`?

Reply:
That makes sense. I think I got that in d9742cb, please take another look.